AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -293,28 +295,41 @@ - (void)displayWillStartAsynchronously:(BOOL)asynchronously
 - (void)didEnterVisibleState
 {
   [super didEnterVisibleState];
-  ASDN::MutexLocker l(__instanceLock__);
+  
+  id downloadIdentifier = nil;
+  
+  {
+    ASDN::MutexLocker l(__instanceLock__);
 
-  if (_downloaderFlags.downloaderImplementsSetPriority) {
-    if (_downloadIdentifier != nil) {
-      [_downloader setPriority:ASImageDownloaderPriorityVisible withDownloadIdentifier:_downloadIdentifier];
+    if (_downloaderFlags.downloaderImplementsSetPriority) {
+      downloadIdentifier = _downloadIdentifier;
     }
   }
   
+  if (downloadIdentifier != nil) {
+    [_downloader setPriority:ASImageDownloaderPriorityVisible withDownloadIdentifier:downloadIdentifier];
+  }
+  
   [self _updateProgressImageBlockOnDownloaderIfNeeded];
 }
 
 - (void)didExitVisibleState
 {
   [super didExitVisibleState];
-  ASDN::MutexLocker l(__instanceLock__);
+  
+  id downloadIdentifier = nil;
 
-  if (_downloaderFlags.downloaderImplementsSetPriority) {
-    if (_downloadIdentifier != nil) {
-      [_downloader setPriority:ASImageDownloaderPriorityPreload withDownloadIdentifier:_downloadIdentifier];
+  {
+    ASDN::MutexLocker l(__instanceLock__);
+    if (_downloaderFlags.downloaderImplementsSetPriority) {
+      downloadIdentifier = _downloadIdentifier;
     }
   }
   
+  if (_downloadIdentifier != nil) {
```

Comment:
Good catch! This was supposed to be downloadIdentifier, not _downloadIdentifier.